quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/StoragePropertyMappers.java

```diff
@@ -32,8 +32,155 @@ public static PropertyMapper[] getMappers() {
+                        .transformer(StoragePropertyMappers::getAreaStorage)
+                        .paramLabel("type")
+                        .build(),
+                fromOption(StorageOptions.STORAGE_ACTION_TOKEN)
+                        .to("kc.spi-action-token-provider")
+                        .mapFrom("storage")
+                        .transformer(StoragePropertyMappers::getCacheStorage)
+                        .paramLabel("type")
+                        .build(),
+                fromOption(StorageOptions.STORAGE_DBLOCK)
+                        .to("kc.spi-dblock-provider")
+                        .mapFrom("storage")
+                        .transformer(StoragePropertyMappers::getDbLockProvider)
+                        .paramLabel("type")
+                        .build(),
+                fromOption(StorageOptions.STORAGE_CACHE_REALM_ENABLED)
+                        .to("kc.spi-realm-cache-default-enabled")
+                        .mapFrom("storage")
+                        .transformer(StoragePropertyMappers::isCacheAreaEnabledForStorage)
+                        .paramLabel(Boolean.TRUE + "|" + Boolean.FALSE)
+                        .build(),
+                fromOption(StorageOptions.STORAGE_CACHE_AUTHORIZATION_ENABLED)
+                        .to("kc.spi-authorization-cache-default-enabled")
+                        .mapFrom("storage")
+                        .transformer(StoragePropertyMappers::isCacheAreaEnabledForStorage)
+                        .paramLabel(Boolean.TRUE + "|" + Boolean.FALSE)
+                        .build(),
+                fromOption(StorageOptions.STORAGE_CACHE_USER_ENABLED)
+                        .to("kc.spi-user-cache-default-enabled")
+                        .mapFrom("storage")
+                        .transformer(StoragePropertyMappers::isCacheAreaEnabledForStorage)
+                        .paramLabel(Boolean.TRUE + "|" + Boolean.FALSE)
+                        .build(),
+                fromOption(StorageOptions.STORAGE_SINGLE_USE_OBJECT)
+                        .to("kc.spi-single-use-object-provider")
+                        .mapFrom("storage")
+                        .transformer(StoragePropertyMappers::getCacheStorage)
+                        .paramLabel("type")
+                        .build(),
         };
     }
+
+    private static Optional<String> getAreaStorage(Optional<String> storage, ConfigSourceInterceptorContext context) {
+        return Optional.of("chm".equals(storage.orElse(null)) ? "map" : "jpa");
+    }
+
+    private static Optional<String> getCacheStorage(Optional<String> storage, ConfigSourceInterceptorContext context) {
+        return Optional.of("chm".equals(storage.orElse(null)) ? "map" : "infinispan");
+    }
+
+    private static Optional<String> getDbLockProvider(Optional<String> storage, ConfigSourceInterceptorContext context) {
+        return Optional.of("chm".equals(storage.orElse(null)) ? "none" : "jpa");
+    }
```

Comment:
I believe the way of checking the `"legacy"` like on line 168 is better than checking for `"chm"` since for any other new storage technology (say JPA or Hot Rod), it will always be `"map"`
```suggestion
    private static Optional<String> getAreaStorage(Optional<String> storage, ConfigSourceInterceptorContext context) {
        return Optional.of("legacy".equals(storage.orElse(null)) ? "jpa" : "map");
    }

    private static Optional<String> getCacheStorage(Optional<String> storage, ConfigSourceInterceptorContext context) {
        return Optional.of("legacy".equals(storage.orElse(null)) ? "infinispan" : "map");
    }

    private static Optional<String> getDbLockProvider(Optional<String> storage, ConfigSourceInterceptorContext context) {
        return Optional.of("legacy".equals(storage.orElse(null)) ? "jpa" : "none");
    }
```

Reply:
Yeah, it makes sense.